src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int GROWTH_RATE = 5;
+    private int size;
+    private T[] elementData;
+    private T[] bufferArray;
```

Comment:
this field actually shouldn't be declared here and could be replaced with local variable